.github/workflows/ci.yml

```diff
@@ -189,13 +189,64 @@ jobs:
         name: coverage
         path: coverage
 
+# We cannot use tracing on 3.12 without Cython 3+, which makes the coverage useless.
```

Comment:
What does "cannot use tracing" on 3.12 mean exactly? Does this have implications for user-facing functionality?

Reply:
I sort of mention this in one of my commit messages, but cython does not support line tracing or profiling for python 3.12+. It definitely affects collecting coverage in our unit tests, but im not aware of how pydantic leverages this otherwise or if its even necessary. I asked for clarification in #9622 about this.

I did want to at least prove that the pydantic package would at least perform a cythonized build and continue to pass tests even of tracing isnt supported.

Reply:
That comment is actually wrong too, because its still broken in cython 3 afaik https://github.com/cython/cython/issues/5470

Reply:
Right I see, so the tradeoff becomes that users of our cythonized build also get broken coverage on 3.12? Or I guess that really the cythonized code is measuring for our internal coverage, and it won't matter to users at all?

Reply:
Yes, I don't know if you ship cythonized builds with tracing enabled or if you just use it to capture coverage for CI purposes.